include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -0,0 +1,310 @@
+//!\cond
+    requires is_type_specialisation_of_v<alignment_configuration_t, configuration>
+//!\endcond
+class pairwise_alignment_algorithm_banded :
+    protected pairwise_alignment_algorithm<alignment_configuration_t, policies_t...>
+{
+protected:
+    //!\brief The alignment configuration traits type with auxiliary information extracted from the configuration type.
+    using base_algorithm_t = pairwise_alignment_algorithm<alignment_configuration_t, policies_t...>;
+
+    // import types from base class.
+    using typename base_algorithm_t::traits_type;
+    using typename base_algorithm_t::scoring_scheme_type;
+    using typename base_algorithm_t::alignment_result_type;
+
+    static_assert(!std::same_as<alignment_result_type, empty_type>, "Alignment result type was not configured.");
+    static_assert(traits_type::is_banded, "Alignment configuration must have band configured.");
+
+    //!\brief The selected lower diagonal.
+    int32_t lower_diagonal{};
+    //!\brief The selected upper diagonal.
+    int32_t upper_diagonal{};
+
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    pairwise_alignment_algorithm_banded() = default; //!< Defaulted.
+    pairwise_alignment_algorithm_banded(pairwise_alignment_algorithm_banded const &) = default; //!< Defaulted.
+    pairwise_alignment_algorithm_banded(pairwise_alignment_algorithm_banded &&) = default; //!< Defaulted.
+    pairwise_alignment_algorithm_banded & operator=(pairwise_alignment_algorithm_banded const &) = default; //!< Defaulted.
+    pairwise_alignment_algorithm_banded & operator=(pairwise_alignment_algorithm_banded &&) = default; //!< Defaulted.
+    ~pairwise_alignment_algorithm_banded() = default; //!< Defaulted.
+
+    /*!\brief Constructs and initialises the algorithm using the alignment configuration.
+     * \param config The configuration passed into the algorithm.
+     *
+     * \details
+     *
+     * Initialises the algorithm given the user settings from the alignment configuration object.
+     *
+     * \throws seqan3::invalid_alignment_configuration if the given band settings are invalid.
+     */
+    pairwise_alignment_algorithm_banded(alignment_configuration_t const & config) : base_algorithm_t(config)
+    {
+        using seqan3::get;
+
+        auto band = get<seqan3::align_cfg::band_fixed_size>(config);
+
+        lower_diagonal = band.lower_diagonal.get();
+        upper_diagonal = band.upper_diagonal.get();
+
+        if (upper_diagonal < lower_diagonal || upper_diagonal < 0 || lower_diagonal > 0)
+            throw invalid_alignment_configuration{to_string("The selected band [",
+                                                            lower_diagonal,
+                                                            ":",
+                                                            upper_diagonal,
+                                                            "] is not supported.")};
```

Comment:
```suggestion
            throw invalid_alignment_configuration{"The selected band [" + std::to_string(lower_diagonal) + ":" + std::to_string(upper_diagonal) + "] is not supported.")};
```

We should avoid `detail::to_string`, because it is really expensive as include.

Reply:
Should we not remove it then, respecitvely deprecate it, or make it just not expensive? Because it is a reall useful tool